library/src/main/java/com/novoda/downloadmanager/Batch.java

```diff
@@ -78,17 +79,26 @@ public Builder(DownloadBatchId downloadBatchId, String title) {
         public Builder addFile(String fileUrl) {
             String rawId = downloadBatchId.rawId() + fileUrl;
             DownloadFileId downloadFileId = DownloadFileIdCreator.createFrom(rawId);
-            fileUrls.put(downloadFileId, fileUrl);
+            NetworkAddressAndFilePath networkAddressAndFilePath = new NetworkAddressAndFilePath(fileUrl, FilePathCreator.unknownFilePath());
+            networkAddressAndFileNameById.put(downloadFileId, networkAddressAndFilePath);
             return this;
         }
 
         public Builder addFile(DownloadFileId downloadFileId, String fileUrl) {
-            fileUrls.put(downloadFileId, fileUrl);
+            NetworkAddressAndFilePath networkAddressAndFilePath = new NetworkAddressAndFilePath(fileUrl, FilePathCreator.unknownFilePath());
+            networkAddressAndFileNameById.put(downloadFileId, networkAddressAndFilePath);
+            return this;
+        }
+
+        public Builder addFile(DownloadFileId downloadFileId, String fileUrl, FilePath filePath) {
+            NetworkAddressAndFilePath networkAddressAndFilePath = new NetworkAddressAndFilePath(fileUrl, filePath);
+            networkAddressAndFileNameById.put(downloadFileId, networkAddressAndFilePath);
```

Comment:
These 3 methods are growing, and are basically just adding params... what about just using `return this.addFile()`s adding each time the new param?